include/boost/geometry/algorithms/simplify.hpp

```diff
@@ -57,7 +61,13 @@ struct simplify_range_insert
     {
         boost::ignore_unused(strategy);
 
-        if (boost::size(range) <= 2 || max_distance < 0)
+        if (boost::size(range) == 2
+            && geometry::equals(geometry::range::front(range),
+                                geometry::range::back(range)))
+        {
+            std::copy(boost::begin(range), boost::begin(range) + 1, out);
+        }
+        else if (boost::size(range) <= 2 || max_distance < 0)
```

Comment:
Do these conditions work the same way for closed and open rings? Shouldn't the size be `3` in case of closed ring? Or is some particular closure assumed?